### PR TITLE
Selenium addon

### PR DESCRIPTION
**Pull request by solker**
#6

### CODE REVIEW

**Comment by solker** [comment]
@amitaibu ready for code review.
One think, if you disable silenium addon in .shoov.yml then you also should remove silenium addon from behat.

File: routes/ci.js [hunk 1]
```diff
@@ -85,17 +114,92 @@ router.get('/:buildItemId/:accessToken', function(req, res, next) {
...
+ *  The owner username of repository.
+ * @param repositoryName
+ *  The repository name.
+ * @param accessToken
+ *  Drupal restful private user token
+ *
+ * @returns {Promise}
+ *  HTTP response in promise wrapper.
+ */
+var getShoovConfig = function(userName, repositoryName, accessToken) {
```
Comment (amitaibu) [inline routes/ci.js]: need to take care of the case that `.shoov.yml` doesn't exist yet-- so we can actually check for it earlier, and if no `.shoov.yml` return early.

File: routes/ci.js [hunk 2]
```diff
@@ -85,17 +114,92 @@ router.get('/:buildItemId/:accessToken', function(req, res, next) {
...
+  var options = {
+    'url': 'https://api.github.com/repos/' + userName + '/' + repositoryName + '/contents/.shoov.yml',
+    'headers': {
+      'Authorization': 'token ' + accessToken,
+      'User-Agent': 'Shoov.io'
```
Comment (amitaibu) [inline routes/ci.js]: why is this needed?

Comment (solker) [inline routes/ci.js]: @amitaibu https://developer.github.com/v3/#user-agent-required

Comment (amitaibu) [inline routes/ci.js]: :+1:

File: routes/ci.js (hunk 1 above)
Comment (solker) [inline routes/ci.js]: What is best approach to fix it? I have no idea, my suggestion are crashed.

**Comment by solker** [comment]
I think we need something like abstract interface to load and parse .shoov.yml to handle all possible errors. And use this interface on php-ci and nodejs backend and other services.

**Comment by amitaibu** [comment]
Please open this as an issue for a follow up, but don't implement for now

File: routes/ci.js [hunk 3]
```diff
@@ -53,11 +55,52 @@ router.get('/:buildItemId/:accessToken', function(req, res, next) {
     }
   };
 
-  // Set the build status to "in progress".
+  var ciBuildItem = undefined;
```
Comment (amitaibu) [inline routes/ci.js]:
```
// 
var ciBuildItem;
```

File: routes/ci.js [hunk 4]
```diff
@@ -53,11 +55,52 @@ router.get('/:buildItemId/:accessToken', function(req, res, next) {
...
     .then(function(response) {
-      var data = JSON.parse(response).data[0];
-      return execDocker(data.build, buildItemId, accessToken);
+      ciBuildItem = JSON.parse(response).data[0];
+      return getUser(accessToken);
+    })
+    // Get user data assigned with CI Build Item.
+    .then(function(response) {
+      userDetail = JSON.parse(response).data[0];
+      return getBuild(ciBuildItem.build, accessToken);
```
Comment (amitaibu) [inline routes/ci.js]: ``// Get the repository name from the Build.``

File: routes/ci.js [hunk 5]
```diff
@@ -53,11 +55,52 @@ router.get('/:buildItemId/:accessToken', function(req, res, next) {
...
+    // Get CI Build data.
+    .then(function(response) {
+      buildDetail = JSON.parse(response).data[0];
+      var userName = buildDetail.label.split('/')[0];
+      var repositoryName = buildDetail.label.split('/')[1];
+      return getShoovConfig(userName, repositoryName, userDetail.github_access_token);
+    })
+    // Get Shoov configuration file from repository.
+    .then(function(response) {
+      if (response === undefined) {
```
Comment (amitaibu) [inline routes/ci.js]: `if (!response) {`

File: routes/ci.js [hunk 6]
```diff
@@ -53,11 +55,52 @@ router.get('/:buildItemId/:accessToken', function(req, res, next) {
...
+      try {
+        var data = JSON.parse(response);
+        var contentDecoded = new Buffer(data.content, 'base64').toString('utf8');
+        var shoovConfig = yaml.safeLoad(contentDecoded);
+      }
+      catch (e) {
+        throw new Error('Invalid .shoov.yml: ' + err.message);
+      }
+
+      if (shoovConfig === undefined) {
```
Comment (amitaibu) [inline routes/ci.js]: `if (!sh..`

**Comment by solker** [comment]
@amitaibu done, merge please.

**Comment by amitaibu** [comment]
Merged, thanks.
